Packs/PAN-OS/Scripts/PanoramaSearchPolicy/PanoramaSearchPolicy.py

```diff
@@ -0,0 +1,67 @@
+import demistomock as demisto  # noqa: F401
+from CommonServerPython import *  # noqa: F401
+from typing import Any
+import traceback
+
+
+''' STANDALONE FUNCTION '''
+
+
+def panorama_search_policy_command(args: Dict[str, Any]) -> list:
+    """Takes rule_name and returns output of !pan-os-list-rules or null message.
+
+    Args:
+        args (Dict[str, Any]): Demisto.args() object.
+
+    Returns:
+        List: Return from !pan-os-list-rules or message.
+    """
+    rule_name = args.get('rule_name', None)
+
+    if not rule_name:
+        raise ValueError('rule_name not specified')
+
+    # Find all device groups + shared.
+    device_groups = []
+    # Need a for loop in case multiple integrations are configured.
+    dg_res = demisto.executeCommand("pan-os-platform-get-device-groups", {})
+    for instance in dg_res:
+        if instance.get('Contents') and not isError(instance):
+            for dev in instance['Contents']:
+                device_groups.append(dev['name'])
+
+    # Error if no device groups found.
+    if len(device_groups) == 0:
+        raise ValueError('Unable to pull device groups')
+
+    # Add "shared" device group
+    device_groups.append('shared')
+
+    # Search for rules
+    pre_post = ['pre-rulebase', 'post-rulebase']
+    for dg in device_groups:
+        for entry in pre_post:
+            response = demisto.executeCommand('pan-os-list-rules', {'device-group': dg, 'pre_post': entry, 'rulename': rule_name})
+            # Need a for loop in case multiple integrations are configured.
+            for res in response:
+                if not isError(res):
+                    return res
+    return ["No matching rule found."]
```

Comment:
1) why do you return a list here?
2) is it possible that you will have multiple rules at the same names under different device-groups?

Reply:
1. I returned a list because the response from !pan-os-list-rules will always be a list and therefor wanted the ouptut of the function to be the same [type](https://github.com/PaloAltoNetworks/xpanse-content/blob/PANOS-EXPANDR-5997/Packs/PAN-OS/Scripts/PanoramaSearchPolicy/PanoramaSearchPolicy.py#L10) always.  I'm open to suggestions.
2. Yes, you are right that the same rule name can be used in different device-groups.  I can make @MosheEichler 's proposed [change](https://github.com/demisto/content/pull/30051#discussion_r1350412352) above to account for that